engine/swarm/stack-deploy.md

```diff
@@ -0,0 +1,250 @@
+
+1.  Start the registry as a service on your swarm:
+
+        $ docker service create --name registry --publish 5000:5000 registry:2
+
+2.  Check its status with `docker service ls`:
+
+        $ docker service ls
+        ID            NAME      REPLICAS  IMAGE                                                                               COMMAND
+        l7791tpuwkco  registry  1/1       registry:2@sha256:1152291c7f93a4ea2ddc95e46d142c31e743b6dd70e194af9e6ebe530f782c17
+
+    Once it reads `1/1` under `REPLICAS`, it's running. If it reads `0/1`, it's
+    probably still pulling the image.
+
+3.  Check that it's working with `curl`:
+
+        $ curl http://localhost:5000/v2/
+        {}
+
+## Create the example application
+
+The app used in this guide is based on the hit counter app in the [Get started
+with Docker Compose](/compose/gettingstarted.md) guide. It consists of a Python
+app which maintains a counter in a Redis instance and increments the counter
+whenever you visit it.
+
+1.  Create a directory for the project:
+
+        $ mkdir stackdemo
+        $ cd stackdemo
+
+2.  Create a file called `app.py` in the project directory and paste this in:
+
+        from flask import Flask
+        from redis import Redis
+
+        app = Flask(__name__)
+        redis = Redis(host='redis', port=6379)
+
+        @app.route('/')
+        def hello():
+            count = redis.incr('hits')
+            return 'Hello World! I have been seen {} times.\n'.format(count)
+
+        if __name__ == "__main__":
+            app.run(host="0.0.0.0", port=8000, debug=True)
+
+3.  Create a file called `requirements.txt` and paste these two lines in:
+
+        flask
+        redis
+
+4.  Create a file called `Dockerfile` and paste this in:
+
+        FROM python:3.4-alpine
+        ADD . /code
+        WORKDIR /code
+        RUN pip install -r requirements.txt
+        CMD ["python", "app.py"]
+
+5.  Finally, create a file called `docker-compose.yml` and paste this in:
+
+        version: '3'
+
+        services:
+          web:
+            image: 127.0.0.1:5000/stackdemo
+            build: .
+            ports:
+              - "8000:8000"
+          redis:
+            image: redis:alpine
+
+    Note that the image for the web app is built using the Dockerfile defined
+    above. It's also tagged with `127.0.0.1:5000` - the address of the registry
+    created earlier. This will be important when distributing the app to the
+    swarm.
+
+
+## Test the app with Compose
+
+1.  Start the app with `docker-compose up`. This builds the web app image,
+    pull the Redis image if you don't already have it, and create two
+    containers.
+
+    You will see a warning about the Engine being in swarm mode. This is because
+    Compose doesn't take advantage of swarm mode, and deploys everything to a
+    single node. You can safely ignore this.
+
+    <!-- TODO: this warning needs to be updated for `docker deploy` -->
```

Comment:
I think this is done: https://github.com/docker/compose/pull/4170/ (needs merged)

Unless you mean something else